src/cli.rs

```diff
@@ -1,14 +1,195 @@
+        .and_then(|metadata| metadata.modified().ok())
+    {
+        None => return true,
+        Some(timestamp) => timestamp,
+    };
+    let config_file_modified_timestamp = config_file_path
+        .metadata()
+        .expect("Couldn't read config file metadata")
+        .modified()
+        .expect("Couldn't get config file modified timestamp");
+    config_file_modified_timestamp > local_release_binary_modified_timestamp
+}
+
+const LOCAL_RULES_DIR_NAME: &str = "local_rules";
+
+fn regenerate_local_binary(
+    local_binary_project_directory: &Path,
+    relative_path_from_local_binary_project_directory_to_project_directory: &Path,
+) {
+    eprintln!("Config changed, regenerating local binary");
+    let parsed_config_file = load_config_file();
+    let local_binary_project_src_directory = local_binary_project_directory.join("src");
+    let local_binary_project_cargo_toml_path = local_binary_project_directory.join("Cargo.toml");
+    if local_binary_project_directory.is_dir() {
+        let _ = fs::remove_dir_all(&local_binary_project_src_directory);
+        let _ = fs::remove_file(&local_binary_project_cargo_toml_path);
+        let _ = fs::remove_file(local_binary_project_directory.join("Cargo.lock"));
+    }
+    fs::create_dir_all(local_binary_project_directory)
+        .expect("Couldn't create local binary project directory");
+    let has_local_rules = local_binary_project_directory
+        .parent()
+        .unwrap()
+        .join(LOCAL_RULES_DIR_NAME)
+        .is_dir();
+
+    let cargo_toml_contents = get_local_binary_cargo_toml_contents(
+        &parsed_config_file,
+        has_local_rules,
+        relative_path_from_local_binary_project_directory_to_project_directory,
+    );
+    fs::write(local_binary_project_cargo_toml_path, cargo_toml_contents)
+        .expect("Couldn't write local binary project Cargo.toml");
+
+    fs::create_dir(&local_binary_project_src_directory)
+        .expect("Couldn't create local binary project `src/` directory");
+    let src_main_rs_contents = get_src_main_rs_contents(&parsed_config_file, has_local_rules);
+    fs::write(
+        local_binary_project_src_directory.join("main.rs"),
+        src_main_rs_contents,
+    )
+    .expect("Couldn't write local binary project src/main.rs");
+
+    let gitignore_contents = get_gitignore_contents();
+    fs::write(
+        local_binary_project_directory.join(".gitignore"),
+        gitignore_contents,
+    )
+    .expect("Couldn't write local binary project .gitignore");
+
+    release_build_local_binary(local_binary_project_directory);
+}
+
+fn release_build_local_binary(local_binary_project_directory: &Path) {
+    let output = Command::new("cargo")
+        .args(["build", "--release"])
+        .current_dir(local_binary_project_directory)
+        .output()
+        .expect("Failed to execute cargo release build command");
+    if !output.status.success() {
+        panic!("Cargo release build of local binary project failed");
```

Comment:
We definitely want to try and make this "friendlier" (since unless we explicitly caught it earlier this is where it would fail on things like "a plugin in your config doesn't actually exist (on crates.io or at the local path specified")

But for now basic failure + "happy path working" is plenty high of a bar